tools/ci/rules.go

```diff
@@ -0,0 +1,23 @@
+//go:build ignore
+// +build ignore
+
+package gorules
+
+import (
+	"github.com/quasilyte/go-ruleguard/dsl"
+)
+
+func fmtErrorfWithoutArgs(m dsl.Matcher) {
+	m.Match(`fmt.Errorf($msg)`).
+		Report(`fmt.Errorf: change for errors.New($msg)`).
+		Suggest(`errors.New($msg)`)
+}
+
+func createHttpClient(m dsl.Matcher) {
+	m.Match(
+		`http.Client{$*_}`,
+		`new(http.Client)`,
+		`http.Transport{$*_}`,
+		`new(http.Transport)`,
+	).Report(`http.Client: use fleethttp.NewClient instead`)
+}
```

Comment:
Love these rules!

Reply:
Yeah that tool is really interesting, didn't know about it before seeing Zach's ticket for this PR. We can certainly leverage it for other rules in the future.